src/mongocxx/exception/error_category.cpp

```diff
@@ -0,0 +1,130 @@
+        case mongoc_error::k_collection_update_failed:
+            return "mongoc collection_update_failed error";
+        case mongoc_error::k_command_invalid_arg:
+            return "mongoc command_invalid_arg error";
+        case mongoc_error::k_cursor_invalid_cursor:
+            return "mongoc cursor_invalid_cursor error";
+        case mongoc_error::k_gridfs_invalid_filename:
+            return "mongoc gridfs_invalid_filename error";
+        case mongoc_error::k_matcher_invalid:
+            return "mongoc matcher_invalid error";
+        case mongoc_error::k_namespace_invalid:
+            return "mongoc namespace_invalid error";
+        case mongoc_error::k_namespace_invalid_filter_type:
+            return "mongoc namespace_invalid_filter_type error";
+        case mongoc_error::k_protocol_bad_wire_version:
+            return "mongoc protocol_bad_wire_version error";
+        case mongoc_error::k_protocol_error:
+            return "mongoc protocol_error error";
+        case mongoc_error::k_protocol_invalid_reply:
+            return "mongoc protocol_invalid_reply error";
+        case mongoc_error::k_query_command_not_found:
+            return "mongoc query_command_not_found error";
+        case mongoc_error::k_query_failure:
+            return "mongoc query_failure error";
+        case mongoc_error::k_query_not_tailable:
+            return "mongoc query_not_tailable error";
+        case mongoc_error::k_scram_not_done:
+            return "mongoc scram_not_done error";
+        case mongoc_error::k_scram_protocol_error:
+            return "mongoc scram_protocol_error error";
+        case mongoc_error::k_server_selection_bad_wire_version:
+            return "mongoc server_selection_bad_wire_version error";
+        case mongoc_error::k_server_selection_failure:
+            return "mongoc server_selection_failure error";
+        case mongoc_error::k_server_selection_invalid_id:
+            return "mongoc server_selection_invalid_id error";
+        case mongoc_error::k_stream_connect:
+            return "mongoc stream_connect error";
+        case mongoc_error::k_stream_invalid_state:
+            return "mongoc stream_invalid_state error";
+        case mongoc_error::k_stream_invalid_type:
+            return "mongoc stream_invalid_type error";
+        case mongoc_error::k_stream_name_resolution:
+            return "mongoc stream_name_resolution error";
+        case mongoc_error::k_stream_not_established:
+            return "mongoc stream_not_established error";
+        case mongoc_error::k_stream_socket:
+            return "mongoc stream_socket error";
+        case mongoc_error::k_unknown:
+        default:
+            return "unknown mongoc error";
+        }
+    }
+};
+
+const std::error_category& mongoc_error_category() {
+    static const mongoc_error_category_impl instance{};
+    return instance;
+}
+
+std::error_code make_mongoc_error_code(mongoc_error code) {
```

Comment:
hm, maybe it would be less confusing if we renamed the "mongoc_error" enum to something like "mongoc_error_index" or "mongoc_error_number."  Then we could use that instead of "mongoc_error_code," along with changing things like this parameter `mongoc_error code` to the corresponding `mongoc_error_index index` instead of using the word 'code.'  Too many name collisions.